pennylane_lightning/src/tests/Test_Measures.cpp

```diff
@@ -163,39 +164,90 @@ TEST_CASE("Expected Values", "[Measures]") {
+                                               0.023377, 0.000471};
+
+    size_t num_qubits = 3;
+    size_t N = std::pow(2, num_qubits);
+    size_t num_samples = 100000;
+    auto &&samples = Measurer.generate_samples(num_samples);
+
+    std::vector<size_t> counts(N, 0);
+    std::vector<size_t> samples_decimal(num_samples, 0);
+
+    // convert samples to decimal and then bin them in counts
+    for (size_t i = 0; i < num_samples; i++) {
+        for (size_t j = 0; j < num_qubits; j++) {
+            if (samples[i * num_qubits + j] != 0) {
+                samples_decimal[i] += twos[(num_qubits - 1 - j)];
+            }
+        }
+        counts[samples_decimal[i]] += 1;
+    }
+
+    // compute estimated probabilities from histogram
+    std::vector<TestType> probabilities(counts.size());
+    for (size_t i = 0; i < counts.size(); i++) {
+        probabilities[i] = counts[i] / (TestType)num_samples;
+    }
+
+    // compare estimated probabilities to real probabilities
+    SECTION("No wires provided:") {
+        REQUIRE_THAT(probabilities,
+                     Catch::Approx(expected_probabilities).margin(.05));
+    }
+}
+
+TEMPLATE_TEST_CASE("Variances", "[Measures]", float, double) {
+    // Defining the State Vector that will be measured.
+    StateVectorManaged<TestType> Measured_StateVector =
+        Initializing_StateVector<TestType>();
+
+    // Initializing the measures class.
+    // It will attach to the StateVector, allowing measures to keep been taken.
+    Measures<TestType, StateVectorManaged<TestType>> Measurer(
+        Measured_StateVector);
 
     SECTION("Testing single operation defined by a matrix:") {
-        vector<std::complex<double>> PauliX = {0, 1, 1, 0};
+        vector<std::complex<TestType>> PauliX = {0, 1, 1, 0};
         vector<size_t> wires_single = {0};
-        double variance = Measurer.var(PauliX, wires_single);
-        double variances_ref = 0.757222;
+        TestType variance = Measurer.var(PauliX, wires_single);
+        TestType variances_ref = 0.757222;
         REQUIRE(variance == Approx(variances_ref).margin(1e-6));
     }
 
     SECTION("Testing single operation defined by its name:") {
         vector<size_t> wires_single = {0};
-        double variance = Measurer.var("PauliX", wires_single);
-        double variances_ref = 0.757222;
+        TestType variance = Measurer.var("PauliX", wires_single);
+        TestType variances_ref = 0.757222;
         REQUIRE(variance == Approx(variances_ref).margin(1e-6));
     }
 
     SECTION("Testing list of operators defined by a matrix:") {
-        vector<std::complex<double>> PauliX = {0, 1, 1, 0};
-        vector<std::complex<double>> PauliY = {0, {0, -1}, {0, 1}, 0};
-        vector<std::complex<double>> PauliZ = {1, 0, 0, -1};
+        vector<std::complex<TestType>> PauliX = {0, 1, 1, 0};
+        vector<std::complex<TestType>> PauliY = {0, {0, -1}, {0, 1}, 0};
+        vector<std::complex<TestType>> PauliZ = {1, 0, 0, -1};
```

Comment:
```suggestion
        vector<std::complex<TestType>> PauliX = {{0, 0}, {1, 0}, {1, 0}, {0, 0}};
        vector<std::complex<TestType>> PauliY = {{0, 0}, {0, -1}, {0, 1}, {0, 0}};
        vector<std::complex<TestType>> PauliZ = {{1, 0}, {0, 0}, {0, 0}, {-1, 0}};
```
I think it is best to be explicit here about each complex value field.